torch/distributed/_composable/fully_shard.py

```diff
@@ -56,7 +57,7 @@ def fully_shard(
     forward_prefetch_limit = 1
     state = _init_core_state(
         state,
-        ShardingStrategy.FULL_SHARD,
+        strategy or ShardingStrategy.FULL_SHARD,
```

Comment:
unrelated to this PR, but we should probably add appropriate documentation to the docstr before open sourcing this / removing the `_`.

Reply:
+1 will need more discussion on the final set of args for `fully_shard`